test/CMakeLists.txt

```diff
@@ -24,6 +24,29 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+function(add_spvtools_unittest target srcs libs)
+  add_executable(${target} ${srcs})
+  spvtools_default_compile_options(${target})
+  if(${COMPILER_IS_LIKE_GNU})
+    target_compile_options(${target} PRIVATE -Wno-undef)
+  endif()
+  if(${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
+    # Disable C4503 "decorated name length exceeded" warning,
+    # triggered by some heavily templated types.
+    # We don't care much about that in test code.
+    # Important to do since we have warnings-as-errors.
+    target_compile_options(${target} PRIVATE /wd4503)
+  endif()
+  target_include_directories(${target} PRIVATE
+    ${spirv-tools_SOURCE_DIR}
+    ${gtest_SOURCE_DIR}/include
+    ${gmock_SOURCE_DIR}/include
+  )
+  target_link_libraries(${target} PRIVATE ${libs})
+  target_link_libraries(${target} PRIVATE gmock_main)
+  add_test(NAME spvtools_${target} COMMAND ${target})
```

Comment:
The test name is inconsistent with the old style. 
Before we had spirv-tools-... and now its spvtools_
I like the hyphen (I think we used that elsewhere too), and I like spelling out spirv-tools.